rclcpp_action/src/server.cpp

```diff
@@ -627,19 +627,19 @@ ServerBase::publish_result(const GoalUUID & uuid, std::shared_ptr<void> result_m
   }
 
   {
-    std::lock_guard<std::recursive_mutex> lock(pimpl_->unordered_map_mutex_);
+    std::lock_guard<std::recursive_mutex> unordered_map_lock(pimpl_->unordered_map_mutex_);
```

Comment:
nit: 

```suggestion
    std::lock_guard unordered_map_lock(pimpl_->unordered_map_mutex_);
```

Reply:
This two changes are inconsistent with other `std::lock_guard` in `server.cpp`.

Reply:
Before, we were using c++14 and this was not possible, so we're going to find inconsistencies.
The idea is to use the less verbose version in new code.

In this case, a backport PR will need to be modified to be compliant with c++14 again.
I think that's fine, but it's also fine if you want to keep the verbose version here.